### DISCUSSION TITLE
mbedTLS Library Support for Azure IoT Middleware FreeRTOS SDK

### DISCUSSION THREAD

**Discussion by MouleshThangavelu** [Azure/azure-iot-middleware-freertos#318] [Q&A]
Hi,

I am looking for the mbedTLS SDK to be used with my STM32H753 microcontroller where I am trying to connect to the azure IoT hub using DPS service. I am trying to integrate core MQTT and mbedTLS into my project with the Azure IoT Middleware FreeRTOS SDK. However, I am not able to identify the right mbedTLS SDK and it's configurations that has to be used to interface with the cloud.

Any help in identifying the appropriate SDK and the mbedTLS config file would be of great help.

Thanks in Advance !!

**Comment by vaavva** [comment]
Hello @MouleshThangavelu !
Here's a sample (using the Middleware for FreeRTOS) that looks like it would be similar to what you're looking for: It uses the STM32H745I-DISCO instead of the STM32H753, but it uses core MQTT, mbedTLS, and supports DPS. 
https://github.com/Azure-Samples/iot-middleware-freertos-samples/tree/main/demos/projects/ST/stm32h745i-disco/cm7

This doc might also have some helpful information: https://github.com/Azure/azure-iot-middleware-freertos/blob/main/docs/azure_iot_c_sdk_port.md

Let me know if this gets you pointed in the right direction or if you have further questions!

**Reply by MouleshThangavelu** [reply]
Hi @vaavva , Thanks for the artifacts shared above. Based on the above samples, I have been integrating the SDKs and currently I am trying to provision my device using the DPS. However, i can see that after the MQTT CONNECT & MQTT CONNACK, the code waits for a response from the server in the AzureIoTMQTT_ProcessLoop. Can you let me know what is expected to be received after that. Also, do we need to do some modification when we are using it with DPS ?
